newrelic-agent/src/main/java/com/newrelic/agent/Transaction.java

```diff
@@ -985,10 +975,6 @@ public TransactionTimer getTransactionTimer() {
     private void finishTransaction() {
         try {
             synchronized (lock) {
-                if (Agent.LOG.isFinestEnabled()) {
-                    threadAssertion();
-                }
```

Comment:
All `threadAssertion()` does is assert that the current thread has a lock on `lock`, so calling it immediately after `synchronized (lock)` is redundant.